### PR TITLE
Use more recent LTS versions on CI instead of the old 16.04

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

File: .ci/azure-pipelines/tutorials.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ jobs:
   - job: tutorials
     displayName: Building Tutorials
     pool:
-      vmImage: 'Ubuntu 16.04'
-    container: env1604
+      vmImage: 'Ubuntu 20.04'
+    container: env1804
```
Comment (larshg) [inline .ci/azure-pipelines/tutorials.yaml:6]: Want to keep the container change in this one?

Comment (kunaltyagi) [reply .ci/azure-pipelines/tutorials.yaml:6]: The tutorials shouldn't run on an image we're no longer interested in, right?

Comment (larshg) [reply .ci/azure-pipelines/tutorials.yaml:6]: No, its just not fit for the headline of the PR :-)

Comment (larshg) [reply .ci/azure-pipelines/tutorials.yaml:6]: But should be moved to another PR?

Comment (kunaltyagi) [reply .ci/azure-pipelines/tutorials.yaml:6]: It's literally a 1 char change. Let me reword the title

**Review by larshg — APPROVED** [review]

**Comment by mvieth** [comment]
Hm, there are multiple instances where `vmImage` _and_ `container` is specified, but the versions do not match. Can that be a problem? Is it even necessary to set both?

**Comment by larshg** [comment]
VMImage is the host OS which run docker.
Container is the actual docker image (ie.where we install all dependencies, fmt, doc, env).
Windows and MacOS doesn't use docker, but just install dependencies directly on host OS.
I have some preliminary work for windows docker images (so we don't end up with current broken CI due boost being changed/updated).

**Review by mvieth — APPROVED** [review]
In `.dev/docker/env/Dockerfile` there is also the line `ARG UBUNTU_DISTRO="16.04"`, but that seems to be overwritten every time.

**Comment by kunaltyagi** [comment]
Good point. I'll update that to 20.04 in the next PR

**Pull request merged**